### DISCUSSION TITLE
Engineering Services and Testing CMDC Policies and Standards

### DISCUSSION THREAD

**Discussion by mcleo-d** [finos/compliant-financial-infrastructure#172] [FINOS and CDMC Roadmap]
# Engineering Services and Testing CMDC Policies and Standards

This discussion has been introduced to start planning how CDMC Policies and Compliance Standards will be engaged, engineered and tested by the [Compliant Financial Infrastructure](https://github.com/finos/compliant-financial-infrastructure/) and [Open Compliance for CDMC](https://github.com/finos/open-compliance-cdmc) project teams.

It has been recognised that there is potential for CDMC Policies and Standards to be engineered at the **Service Level**, i.e. individual services created through Terraform and other IaC means, and at the **CSP Infrastructure Level**, which is out reach from engineers and requires the CSPs to configure directly.

The following diagrams illustrate the relationship between CDMC and FINOS projects, alongside two diagrams that explain the scenarios above and which propose two potential routes forwards. These potential routes are open for debate, modification and complete reshaping in this discussion. 

<img src="https://user-images.githubusercontent.com/6029572/142853715-ccb86cc0-4117-4089-89c1-4d83edfd7b3a.png" width="70%">

### Summary
- Cloud service IaC is developed according to Compliant Financial Infrastructure prioritised backlog.
- Probr tests run against individual cloud service providers.
- BDD results aggregated in API reporting system. See Scenario 2.

<img src="https://user-images.githubusercontent.com/6029572/142853729-aac4938c-5023-4de2-bb0f-7d9e5dd0749c.png" width="70%">

### Summary
- Cloud service providers register their own CDMC compliance against a FINOS / LF hosted reporting API.
- Reports are aggregate with Probr test results from IaC service development and testing.
- Reporting and API System to be determined.

<img src="https://user-images.githubusercontent.com/6029572/142853725-7dcfa50e-146a-42ab-ba56-c62d9608acf4.png" width="70%">

**Comment by noelmcloughlin** [comment]
Looking at the diagram, CDMC TSWG produce "_detailed technical spec for compliance tests_".  
Usually standards groups issue these as PDF specifications.
What artifacts are in the "BDD Compliance test pack"? 
Are they machine readable and agnostic to Compliance Check Service implementation? 
I see Terraform specifically mentioned, but to avoid design bias, I'd emphasize "IAC" more generally to avoid decoupling to tools.

**Reply by mcleo-d** [reply]
Hi @noelmcloughlin - Thanks so much for your questions and feedback. Let me pull @eddie-knight, @peterrhysthomas and @abdullahgarcia into the conversation alongside @CrypticFrog and @gibsocol 👍🏻

**Reply by mcleo-d** [reply]
Hi @noelmcloughlin - I have added this discussion to the Compliant Financial Infrastructure agenda for further discussion on Friday 3rd December at 3pm GMT / 10am ET if you would like to join the call ... https://github.com/finos/compliant-financial-infrastructure/issues/173

**Reply by eddie-knight** [reply]
> What artifacts are in the "BDD Compliance test pack"?

All artifacts produced by the test packs will be created by a shared SDK, which will ensure that they are harmonized and machine readable independent of which pack is being run.

**Comment by tcraigjohnson** [comment]
As Friday was my first time with the call I have some questions, please.  

- What are the time scales for finishing the project/tool set itself and the standards?

- Are the standards based on any existing standards like AWS CIS or other similar?

- How do regulators (whether internal or external) want to consume the verification of compliance?

Thanks in advance.

Tj

**Reply by mcleo-d** [reply]
Hi @tcraigjohnson - Thanks so much for your questions. Let me pull @eddie-knight, @peterrhysthomas and @abdullahgarcia into the conversation alongside @CrypticFrog and @gibsocol 👍🏻

**Reply by mcleo-d** [reply]
Hi @tcraigjohnson - I have added this discussion to the Compliant Financial Infrastructure agenda for further discussion on Friday 3rd December at 3pm GMT / 10am ET if you would like to join the call ... https://github.com/finos/compliant-financial-infrastructure/issues/173

**Comment by mcleo-d** [comment]
This is the answer to this issue @thinkl33t
